core/arch/arm/plat-synquacer/timer_fiq.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#include <arm.h>
+#include <console.h>
+#include <drivers/gic.h>
+#include <io.h>
+#include <kernel/panic.h>
+#include <kernel/misc.h>
+#include <kernel/spinlock.h>
+#include <timer_fiq.h>
+
+static unsigned int timer_lock = SPINLOCK_UNLOCK;
+static bool timer_fiq_running;
+
+void generic_timer_start(void)
```

Comment:
Minor: I find the name a bit weird, I mean "generic" although this is platform code.
If it is really generic (and therefore could be used by other platforms), then this implementation file should be elsewhere (`core/arch/arm/kernel` probably). If it is not, then "generic" should be omitted.

Reply:
Yes this is based on Secure EL1 physical timer (D6.1.5 Timers ARNv8-A RM). But currently it fires at fixed duration of 2ms. I think this could be made configurable as an argument (time in ms) to timer APIs. Apart from that I think its good to go in general code as `core/arch/arm/kernel/timer_a64.c`. Please share your views.

Reply:
@b49020 sounds good to me